ivadomed/models.py

```diff
@@ -1440,6 +1441,80 @@ def forward(self, x):
         return net
 
 
+class MONAIUNet(Module):
+    """MONAI's BasicUNet (2D/3D) model adopted in ivadomed"""
+    def __init__(self, in_channel, out_channel, features=(32, 32, 64, 128, 256, 32),
+                 attention=False, dropout_rate=0.3, final_activation="sigmoid", **kwargs):
+        super(MONAIUNet, self).__init__()
+        self.in_channel = in_channel
+        self.out_channel = out_channel
+        self.features = features
+        self.attention = attention
+        self.dropout_rate = dropout_rate
+        self.final_activation = final_activation
+
+        ivadomed_params = {**self.__dict__, **kwargs}
+        monai_model_keys = ['features', 'act', 'norm', 'bias', 'upsample']
+        monai_params = ivadomed_params_to_monai_params(ivadomed_params, monai_model_keys)
+
+        self.model = basic_unet.BasicUNet(**monai_params)
+
+    def forward(self, x):
+        out = self.model(x)
+
+        if hasattr(self, "final_activation") and self.final_activation not in ["softmax", "relu", "sigmoid"]:
+            raise ValueError("final_activation value has to be either softmax, relu, or sigmoid")
+        elif hasattr(self, "final_activation") and self.final_activation == "softmax":
+            out = self.softmax(out)
+        elif hasattr(self, "final_activation") and self.final_activation == "relu":
+            out = nn.ReLU()(out) / nn.ReLU()(out).max() if bool(nn.ReLU()(out).max()) else nn.ReLU()(out)
```

Comment:
Why this division by the maximum value of the output? I know it was in ivadomed's Unet code, but what's the original motivation for it?

Reply:
See [the docs on soft GT](https://ivadomed.org/configuration_file.html#soft-gt) for more context but essentially we use a "normalized" version of `relu` as the final activation to keep values in the range `0` and `1` (i.e., typical range for segmentation).